### PR TITLE
tweaks to make rust-analyzer happy

### LINKED ISSUE #1952: Make Project Compatible With Rust Analyzer
The current project structure does not work with rust analyzer.  This makes it hard to develop using various open-source IDEs such as *emacs* and *neovim*.

### PR DESCRIPTION
**Pull request by dwerner**
resolves #1952

### CODE REVIEW

File: execution_engine/src/core/runtime/auction_internal.rs [hunk 1]
```diff
@@ -118,7 +118,7 @@ where
     }
 
     fn is_allowed_session_caller(&self, account_hash: &AccountHash) -> bool {
-        self.is_allowed_session_caller(account_hash)
+        Runtime::is_allowed_session_caller(self, account_hash)
```
Comment (xcthulhu) [inline execution_engine/src/core/runtime/auction_internal.rs:121]: 👍

**Review by xcthulhu — APPROVED** [review]
LGTM

**Comment by dwerner** [comment]
bors try

**Pull request merged**